services/motion/builtin/builtin.go

```diff
@@ -231,24 +236,111 @@ func (ms *builtIn) MoveOnGlobe(
 	kinematicsOptions.HeadingThresholdDegrees = 8
 
-	plan, kb, err := ms.planMoveOnGlobe(
-		ctx,
-		componentName,
-		destination,
-		movementSensorName,
-		obstacles,
-		kinematicsOptions,
-		extra,
-	)
-	if err != nil {
-		return false, fmt.Errorf("error making plan for MoveOnMap: %w", err)
-	}
+	movementSensor, ok := ms.movementSensors[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// TODO: if we disentangle creation of the kinematic base from the plan call we make we don't need this and can do this once
+	var kinematicBase kinematicbase.KinematicBase = nil
+
+	// Shared variables for planning and execution threads, are all guarded with a mutex
+	var planMu sync.Mutex
+	var moveCtx context.Context
+	var currentPlan [][]referenceframe.Input = nil
+
+	// Shared variable between threads that can trigger replanning and planning thread
+	var replanFlag atomic.Bool
+	replanFlag.Store(true)
+
+	var successFlag atomic.Bool
+	successFlag.Store(false)
+	errChan := make(chan error)
+
+	ms.backgroundWorkers.Add(1)
+	goutils.ManagedGo(func() {
+
+		// maybe move this into the restart tickers function
+		cancelCtx, cancelFn := context.WithCancel(context.Background())
+		ms.cancelFn = cancelFn
+
+		positionPollingTicker := time.NewTicker(time.Duration(1000/motionCfg.PositionPollingFreqHz) * time.Millisecond)
+		obstaclePollingTicker := time.NewTicker(time.Duration(1000/motionCfg.ObstaclePollingFreqHz) * time.Millisecond)
+
+		for !successFlag.Load() && ctx.Err() == nil {
+			select {
+			case <-positionPollingTicker.C:
+				// TODO: theres nothing stopping us from making another thread before the previous has completed
+				ms.backgroundWorkers.Add(1)
+				goutils.ManagedGo(func() {
+					fmt.Println("position poll")
+					current, _, err := movementSensor.Position(cancelCtx, nil)
+					if err != nil {
+						errChan <- err
+						return
+					}
+					if spatialmath.GeoPointToPose(current, destination).Point().Norm() <= kinematicsOptions.PlanDeviationThresholdMM {
+						successFlag.Store(true)
+					}
+
+					// TODO: the function that actually monitors position - and pass context in
+				}, ms.backgroundWorkers.Done)
+			case <-obstaclePollingTicker.C:
+				// TODO: theres nothing stopping us from making another thread before the previous has completed
+				ms.backgroundWorkers.Add(1)
+				goutils.ManagedGo(func() {
+					fmt.Println("obstacle poll")
+					replanFlag.Store(true)
+
+					// this is gonna get removed when I pass context along the way
+					if !utils.SelectContextOrWait(cancelCtx, 1*time.Millisecond) {
+						fmt.Println("I've been cancelled")
+					}
+
+					// TODO: the function that actually monitors obstacles - and pass context in
+				}, ms.backgroundWorkers.Done)
+			default:
+			}
+			if replanFlag.Load() {
+				fmt.Println("replanning")
+
+				ms.cancelFn()
+				cancelCtx, cancelFn = context.WithCancel(context.Background())
```

Comment:
I am using the background context so that the MoveOnGlobe call can exit from the function without waiting on all the threads it spawned.  The motion service owns the waitgroup and the cancel functions and will call them on cleanup.  

We still would be able to cancel the main moveOnGlobe call by cancelling the passed in context.  This would temporarily orphan the threads created inside this loop, but since the cancellation would be done by closing the motion service I think this is acceptable

Reply:
I see. I think we should at least call cancelFn if the MoveOnGlobe call context is canceled.

Reply:
> I am using the background context so that the MoveOnGlobe call can exit from the function without waiting on all the threads it spawned.

Why do we want to exit without waiting on spawned threads? And why is a background context necessary here?

As far as I can tell, what is described above is also not what will happen. If `ctx` is cancelled after a plan begins to be executed, the `MoveOnGlobe` function will continue with its `GoToInputs` calls and *not* exit until something also calls `cancelFunc`.

Reply:
Oh, actually I'm really confused. Since cancelFn is stored in the motion service, calling cancelFn will cancel the threads for all MoveOnGlobe calls. How can we cancel just the threads for this call?

Reply:
Thanks to the op manager I believe that subsequent MoveOnGlobe calls will cancel any prior ones anyway

Reply:
Will it cancel any MoveOnGlobe call? I would hope it only cancels MoveOnGlobe calls for the same base.

Reply:
Yes it would cancel the active MoveOnGlobe call for the motion service, which could be acting on any base.

Reply:
@raybjork and I talked offline that both the op manager and motion service are written expecting a single call at a time to the motion service, so if we ever wanted to change that, there would be significant changes to both of these. Also, @raybjork said the argument for letting the call to exit without waiting on spawned threads is in case the motion service is calling an externally defined vision service that doesn't respect cancellation. That all makes sense to me.

I still think that canceling a request should attempt to cancel the in-progress threads, even if it doesn't wait for cancelation though. We shouldn't wait until the motion service is close.